test/adapter/json/collection_test.rb

```diff
@@ -85,6 +85,72 @@ def test_include_option
 
             assert_equal(expected, actual)
           end
+
+          def test_fields_option
+            serializer = ArraySerializer.new([@first_post, @second_post])
+            adapter = ActiveModel::Serializer::Adapter::Json.new(serializer)
+            actual = adapter.serializable_hash(fields: [:id])
```

Comment:
(that would be a better test usage in the integration test, where the options to `render json: foo, ....` are passed to AMS then to the renderer (unless we mutate them))

@vasilakisfil Are you still interested in this?